src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -370,6 +379,17 @@ public void regenerateIsland(@NotNull Island island, User user, @NotNull Schemat
         return IridiumSkyblock.getInstance().getDatabaseManager().getUserTableManager().getEntries(island);
     }
 
+
+    /**
+     * Gets a list of all Members on an island
+     *
+     * @param island The Specified Island
+     * @return A list of all members on this island
+     */
+    public @NotNull List<User> getPlayersOnIsland(@NotNull Island island) {
+        return Bukkit.getOnlinePlayers().stream().filter(player -> island.isInIsland(player.getLocation())).map(player -> IridiumSkyblock.getInstance().getUserManager().getUser(player)).collect(Collectors.toList());
```

Comment:
this will be much laggy too, (with 100+ players and there 5 players opens visitors gui)

Reply:
Not really, I tested it with 500 players, and it took 1.1ms to complete the list
https://imgur.com/bh2Ewhh.png

Reply:
And with parallel streams it only takes 0.6ms to complete with 500 players online

Reply:
u must open this gui with 5 players, we wont more players, more guis -,-